examples/thermostat/silabs/src/TemperatureManager.cpp

```diff
@@ -92,26 +98,67 @@ void TemperatureManager::AttributeChangeHandler(EndpointId endpointId, Attribute
         int8_t Temp = ConvertToPrintableTemp(*((int16_t *) value));
         SILABS_LOG("Local temp %d", Temp);
         mCurrentTempCelsius = Temp;
+#ifdef DIC_ENABLE
```

Comment:
All of this should not be there. Temperature Manager shouldn't have protocol specific stuff in it. Try the AppTask to hook these api inside.